jrnl/util.py

```diff
@@ -180,9 +196,58 @@ def get_text_from_editor(config, template=""):
     return raw
 
 
-def colorize(string):
-    """Returns the string wrapped in cyan ANSI escape"""
-    return u"\033[36m{}\033[39m".format(string)
+def colorize(string, color, bold=False):
+    """Returns the string colored with colorama.Fore.color. If the color set by
+    the user is "NONE", it returns the string without any modification. Otherwise,
+    If the color set by the user doesn't exist in the colorama.Fore attributes,
+    the colorization is done with WHITE."""
+    upper_color = color.upper()
+    if upper_color == "NONE":
+        return string
+
+    color_escape = getattr(colorama.Fore, upper_color, colorama.Fore.WHITE)
+    if not bold:
+        return color_escape + string + colorama.Fore.RESET
+    else:
+        return colorama.Style.BRIGHT + color_escape + string + colorama.Style.RESET_ALL
+
+
+def highlight_tags_maintain_background_color(entry, text, color, bold=False):
+    """
+    Takes a string and colorizes the tags in it based upon the config value for
+    color.tags, while colorizing the rest of the text based on color.
+    :param entry: Entry object, for access to journal config
+    :param text: Text to be colorized
+    :param color: Color for non-tag text, passed to colorize()
+    :param bold: Bold flag text, passed to colorize()
+    :return: Colorized str
+    """
+    config = entry.journal.config
+    if config['highlight']:  # highlight tags
+        if entry.journal.search_tags:
```

Comment:
The code duplication here bothers me but I don't know how to clean it any more at the moment.

Reply:
Figured out how to condense this, but caused a regression in the test for unreadable dates that I'm working on fixing at the moment.